lib/runner.js

```diff
@@ -795,7 +795,7 @@ Runner.prototype.run = function (fn) {
   var rootSuite = this.suite;
 
   // If there is an `only` filter
-  if (this.hasOnly) {
+  if (hasOnly(rootSuite)) {
```

Comment:
Worth noting that #2874 also touches `hasOnly`-using code in addition to the `--forbid-only` code already merged doing so.

(I plan on a more detailed review later this week.)